src/FakeItEasy/Core/FakeManager.PropertyBehaviorRule.cs

```diff
@@ -55,7 +55,7 @@ public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
             private static PropertyInfo GetProperty(MethodInfo propertyGetterOrSetter)
             {
                 return
-                    (from property in propertyGetterOrSetter.DeclaringType.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                    (from property in propertyGetterOrSetter.DeclaringType!.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)

```

Comment:
DeclaringType can only be null for non-nested types, module-level methods, or dynamic methods.